### PR TITLE
Fix issue 32. Reorganized Service class to provide clients in props

### PR DESCRIPTION
**Pull request by vitali-sonchyk-epam**

### CODE REVIEW

**Comment by nvborisenko** [comment]
@aliaksandrbasau Alex, your opinion is strongly appreciated.

What is better:
- `service.LaunchClient.GetLaunchAsync(id)`
- `service.Launch.GetAsync(id)`

This PR is almost about restructuring classes to split them by namespaces to be possible easy increase api coverage. Since we are breaking everything here, don't hesitate to break everything.

We need design client library very well to avoid breaking changes in future. One more thing I keep in mind - api versioning. What if server will support `/api/v2` endpoint, how we are going to support it.

**Comment by aliaksandrbasau** [comment]
@nvborisenko I wouldn't use `Client` in class and property names as those classes map to REST collections (or resources). I will give it more thoughts later in the evening and give you more detailed answer.

**Comment by aliaksandrbasau** [comment]
@nvborisenko 
Actually `Client` may also work (at least for class names) if those clients are intended to be used separately from the Service class. They map to what Report Portal API documentation calls as `controllers`. So they can be some kind of `ControllerClient`s.
And I would rename `Service` class to `Client` or `ServiceClient` to be more precise about the purpose of the class.

Another way to answer the question is to check how other projects call similar types of classes. E.g. [Swagger CodeGen](https://github.com/swagger-api/swagger-codegen) that can generate c# client from Swagger file. It generates one `ApiClient` class (that is responsible for serialization, deserialization, making requests) and bunch of `Api` classes (one per controller) that consume `ApiClient` and talk to controllers via `ApiClient`.
image

As for `LaunchClient` vs `Launch`, I would go with `Launch` because almost all properties under `Service` as clients.

As for `GetLaunchAsync` vs `GetAsync`, `GetLaunchAsync` might be a better choice since the name makes it clear what it "gets".

To sum up:
`class LaunchClient` => `class LaunchControllerClient`
`class Service` => `ServiceClient` or `ReportPortalServiceClient`
`Service.LaunchClient` => `Service.Launch`
`GetLaunchAsync`

**Comment by vitali-sonchyk-epam** [comment]
@aliaksandrbasau , do u have a suggestions regarding the api versioning?
currently the api prefix is hardcoded in Service constructor.

**Comment by aliaksandrbasau** [comment]
@vitali-sonchyk-epam Yep, forgot to comment on versioning.
Currently version prefix is storied in `ReportPortal.config.json` (e.g. `https://rp.epam.com/api/v1/`).
If the prefix is moved into the client (hard-coded like `BaseUri.Append($"v1/{Project}/launch")`), the client would have full control over what endpoints to use and end-users would make less mistakes by specifying wrong API version (as long as they know not to put version into API url in the config file). And we would be able to use different version of API for different requests (if needed) and we could migrate the client to a newer version of API without any effort on end-users' side.
The only downsides of the approach are that the compatibility is broken and that end-users have to be notified about the changes in URL format.

**Comment by nvborisenko** [comment]
Thanks all, further work is coming separately.
